tiledb/libtiledb.pyx

```diff
@@ -4409,7 +4409,9 @@ cdef class Query(object):
             for dname in dims:
                 if not domain.has_dim(dname):
                     raise TileDBError("Selected dimension does not exist: '{name}")
-            self.dims = dims
+
+            from six import u as unicode 
+            self.dims = [unicode(dname) for dname in dims]
```

Comment:
You can use `dname.encode("UTF-8")` instead to avoid the import. (I'm a little surprised this was necessary, I guess because it is user input so might be bytes?)

Reply:
OK will do. I get [this](https://github.com/TileDB-Inc/TileDB-Py/runs/1591496316?check_suite_focus=true#step:9:425) error with Python 2.